src/LevelLoader.cpp

```diff
@@ -0,0 +1,298 @@
+
+LevelLoader::~LevelLoader() {}
+
+
+/* Open for reading only for now, if we want to save gamestate we can at
+   some point. Possible saveLevel() member function? */
+int LevelLoader::loadLevel(GameWorld &world, std::shared_ptr<GameObject> &player) {
+   std::ifstream level_file("../levels/grandmas-awakening.json",
+      std::ifstream::in);
+   int res = 0;
+
+   if (!level_file) {
+      std::cerr << "Error reading JSON level file." << std::endl;
+      return res;
+   }
+   json level;
+   level_file >> level;
+
+   std::cout << "level_name: " << level["level-name"] << std::endl;
+
+   if ((res = parseShaders(level["level-shaders"]))) {
+      std::cerr << "Error parsing isomorphic shaders." << std::endl;
+      return res;
+   }
+
+   if ((res = parseShapes(level["level-shapes"]))) {
+      std::cerr << "Error parsing shapes." << std::endl;
+      return res;
+   }
+
+   if ((res = parseMaterials(level["level-materials"]))) {
+      std::cerr << "Error parsing materials." << std::endl;
+      return res;
+   }
+
+   if ((res = parseCharacters(world, player, level["characters"]))) {
+      std::cerr << "Error parsing characters." << std::endl;
+      return res;
+   }
+
+   if ((res = parseStaticObjects(world, level["static-objects"]))) {
+      std::cerr << "Error parsing static objects." << std::endl;
+      return res;
+   }
+
+   if ((res = parseDynamicObjects(world, level["dynamic-objects"]))) {
+      std::cerr << "Error parsing dynamic objects." << std::endl;
+      return res;
+   }
+
+   return res;
+}
+
+int LevelLoader::parseShaders(json shaders) {
+   ResourceManager& resourceManager = ResourceManager::instance();
+
+   if (shaders != nullptr) {
+      ShaderManager& shaderManager = ShaderManager::instance();
+
+      for (json shader : shaders) {
+         if (shaderManager.createIsomorphicShader(resourceManager,
+             shader["name"], shader["file-prefix"]) == 0) {
+            return 1;
+         }
+      }
+   }
+
+   return 0;
+}
+
+int LevelLoader::parseShapes(json shapes) {
+   ResourceManager& resourceManager = ResourceManager::instance();
+
+   if (shapes != nullptr) {
+      ShapeManager& shapeManager = ShapeManager::instance();
+
+      for (json shape : shapes) {
+         if (shapeManager.createShape(resourceManager, shape["name"],
+             shape["filename"])) {
+            return 1;
+         }
+      }
+   }
+
+   return 0;
+}
+
+int LevelLoader::parseMaterials(json materials) {
+   if (materials != nullptr) {
+      MaterialManager& materialManager = MaterialManager::instance();
+
+      for (json mat : materials) {
+         std::shared_ptr<Material> material = std::make_shared<Material>();
+         *material = {
+            mat["ambient"]["r"], mat["ambient"]["g"], mat["ambient"]["b"],
+            mat["diffuse"]["r"], mat["diffuse"]["g"], mat["diffuse"]["b"],
+            mat["specular"]["r"], mat["specular"]["g"], mat["specular"]["b"],
+            mat["shininess"]
+         };
```

Comment:
You can't do it on `struct` types, I could either add a constructor to the struct do this
```
std::shared_ptr<Material> material = std::make_shared<Material>(Material{
   mat["ambient"]["r"], mat["ambient"]["g"], mat["ambient"]["b"],
   mat["diffuse"]["r"], mat["diffuse"]["g"], mat["diffuse"]["b"],
   mat["specular"]["r"], mat["specular"]["g"], mat["specular"]["b"],
   mat["shininess"]
});
```
But that's already pretty similar. Or we can make a class for it? Idk if it's worth the effort though, what do you think?